src/rpm/headers/types.rs

```diff
@@ -201,9 +173,37 @@ pub struct FileOptions {
     pub(crate) caps: Option<FileCaps>,
 }
 
+// @todo: should we even have "default permissions" / mode, or use Option<FileMode>?
+// if they can be skipped (unsure), 'inherit_permissions' could go away
 impl FileOptions {
+    /// Create a new FileOptions for a regular file
+    #[allow(clippy::new_ret_no_self)]
+    pub fn regular(dest: impl Into<String>) -> FileOptionsBuilder {
+        FileOptionsBuilder {
+            inner: FileOptions {
+                destination: dest.into(),
+                user: "root".to_string(),
+                group: "root".to_string(),
+                symlink: "".to_string(),
+                mode: FileMode::regular(0o664),
+                flag: FileFlags::empty(),
+                inherit_permissions: true,
+                caps: None,
+            },
+        }
+    }
+
+    /// Create a new FileOptions for a directory
+    ///
+    /// Used to explicitly own the directory itself but not it's contents. This does NOT
+    /// process any of the contents of a directory.
     #[allow(clippy::new_ret_no_self)]
-    pub fn new(dest: impl Into<String>) -> FileOptionsBuilder {
```

Comment:
The basic principle behind the change is that symlinks need to include the destination where they link to, and the initial PR had the user provide that information with a `.symlink()`, which created ambiguity.

However, on second thought, maybe instead of having the user specify this, we should assume the file being pointed to always exists (in practice maybe we do anyway) and use `std::fs::read_link` to get that info?

Reply:
@syokensyo ^ wdyt, since this impacts how we handle symlinks

Reply:
I think it is better to make sure the `link` exists in the build root, and use `std::fs::read_link` read the `link`'s value as the target, however it is also better to provide a custom `link target` api with high priority over `link`'s value.

Reply:
Leaving notes for myself

* yes `rpm` always tries to look for the files and get `stat` info, but in the case of `%ghost` files they aren't required to exist and they literally create fake stat data
* they use `lstat` which I believe is equivalent to https://doc.rust-lang.org/std/fs/fn.symlink_metadata.html
* I think `%attr` can't be applied to symlinks?

see rpm/build/files.c

Does it really make sense to provide a link override though?  If all the other files have to exist, why make a totally separate codepath?

Reply:
"provide a link override" is flow original behavior... and yes, you are right, there is no scenario for that kind of behavior if other files exists.